File: order-service/src/main/java/com/paulopacifico/orderservice/messaging/api/PaymentSucceededEvent.java
package com.paulopacifico.orderservice.messaging.api;

import java.math.BigDecimal;
import java.time.OffsetDateTime;
import java.util.UUID;

public record PaymentSucceededEvent(
        UUID eventId,
        Long orderId,
        String orderNumber,
        BigDecimal amount,
        OffsetDateTime occurredAt
) {
}

File: payment-service/src/main/kotlin/com/paulopacifico/paymentservice/messaging/api/PaymentSucceededEvent.kt
package com.paulopacifico.paymentservice.messaging.api

import java.math.BigDecimal
import java.time.OffsetDateTime
import java.util.UUID

data class PaymentSucceededEvent(
    val eventId: UUID,
    val orderId: Long,
    val orderNumber: String,
    val amount: BigDecimal,
    val occurredAt: OffsetDateTime,
)
